src/test/java/roomescape/payment/application/PaymentServiceTest.java

```diff
@@ -0,0 +1,150 @@
+package roomescape.payment.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import roomescape.common.exception.impl.BadRequestException;
+import roomescape.payment.application.dto.PaymentConfirmRequest;
+import roomescape.payment.application.dto.PaymentDataRequest;
+import roomescape.payment.domain.Payment;
+import roomescape.payment.domain.PaymentStatus;
+import roomescape.reservation.domain.Reservation;
+
+@SpringBootTest
+@ActiveProfiles("test")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@Import(PaymentServiceTest.TestConfig.class)
+class PaymentServiceTest {
+
+    @Autowired
+    private PaymentService paymentService;
+
+    @Autowired
+    private PaymentClient paymentClient;
+
+    @TestConfiguration
+    static class TestConfig {
+        @Bean
+        @Primary
+        public PaymentClient paymentClient() {
+            return mock(PaymentClient.class);
+        }
+    }
```

Comment:
```suggestion
    @MockitoBean
    private PaymentClient paymentClient;
```
이렇게 사용해 볼 수 있지 않을까요?

Reply:
스프링 부트에서 `@MockBean`이 deprecated 된 줄로 알고 있었는데, 
스프링 프레임워크 6.2부터 `@MockitoBean`이 새롭게 추가되었군요! 반영해볼게요